root/account/applications/RegisterForm.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+import noop from 'lodash/noop';
+
+import FormRow from '../../components/FormRow';
+import FormRowSelect from '../../components/FormRowSelect';
+import FormRowText from '../../components/FormRowText';
+import FormRowURLLong from '../../components/FormRowURLLong';
+import FormSubmit from '../../components/FormSubmit';
+import {addColon, l, N_l} from '../../static/scripts/common/i18n';
+import {Lens, prop, set, compose3} from '../../static/scripts/common/utility/lens';
+import getSelectValue from '../../utility/getSelectValue';
+import hydrate from '../../utility/hydrate';
+
+type OauthTypeT = 'installed' | 'web';
+
+type RegisterApplicationFormT = FormT<{|
+  +name: FieldT<string>,
+  +oauth_redirect_uri: FieldT<string>,
+  +oauth_type: FieldT<OauthTypeT>,
+|}>;
+
+type Props = {|
+  +form: RegisterApplicationFormT,
+|};
+
+type State = {|
+  form: RegisterApplicationFormT,
+|};
+
+const oauthTypeOptions = {
+  grouped: false,
+  options: [
+    {label: N_l('Web Application'), value: 'web'},
+    {label: N_l('Installed Application'), value: 'installed'},
+  ],
+};
+
+const oauthTypeFieldLens: Lens<RegisterApplicationFormT, OauthTypeT> =
+  compose3(prop('field'), prop('oauth_type'), prop('value'));
+
+class RegisterApplicationForm extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      form: set(
+        oauthTypeFieldLens,
+        (getSelectValue(
```

Comment:
Not sure I understand this bit. Shouldn't the correct value be in `props.form` already so you can just set `this.state = {form: props.form}`?

Reply:
It should, but it wasn’t.
It is now set from https://github.com/metabrainz/musicbrainz-server/pull/708/commits/5220b4f3c742e02924f2594db2fa497d638a6609#diff-5053a22c07db9b62eceff446a9b99731R710
Note that other fields are set with default value `''` as React is unhappy with `null`.